autoload/lsp/diag.vim

```diff
@@ -141,6 +154,24 @@ def DiagsRefresh(bnr: number)
     return
   endif
 
+  # Initialize default/fallback properties for diagnostic virtual text:
+  var diag_align: string = 'above'
+  var diag_wrap: string = 'truncate'
+  var diag_symbol: string = '┌─'
+  var diag_padding: number = 0
+
+  if opt.lspOptions.diagVirtualTextAlign ==? 'below'
```

Comment:
`==?` matches case insensitive, making `'BeLOw' ==? 'below', is this intentional?

Reply:
Yes this was my intention to be more resilient against typos made by the user. Do you see any drawbacks?

Reply:
Only that `lsp-opt-completionMatcher` doesn't work like that. But we could make it work that way.  😄

Reply:
I'm the newbie here in this code base and have to adhere to the legacies  :smile:
Therefore I leave the decision to you and @yegappan  :smile:

Reply:
The option value comparison should be case sensitive.  Can you use `==#` instead of `==?`?

Reply:
> I'm the newbie here in this code base and have to adhere to the legacies smile Therefore I leave the decision to you and @yegappan smile

as you said, it had no drawbacks, 
but just need to respect/keep the code style.
// which normally should be a common sense in the most open-source in the world, 

HOWEVER this is not the way here in this code base, @yegappan can break/revert their own created style/rule in anytime, and amazedly preferred to listen to someone no sense no why.......
but good news this is a open source, you can modify your copy/fork anytime, unless you want/like/thought it is worthy to share it with someone else then a PR.